pkg/resources/aws/eni.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"k8s.io/klog/v2"
+
+	"k8s.io/kops/pkg/resources"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+)
+
+func DeleteENI(cloud fi.Cloud, r *resources.Resource) error {
+	c := cloud.(awsup.AWSCloud)
+
+	id := r.ID
+
+	klog.V(2).Infof("Deleting EC2 ENI %q", id)
+	request := &ec2.DeleteNetworkInterfaceInput{
+		NetworkInterfaceId: &id,
+	}
+	_, err := c.EC2().DeleteNetworkInterface(request)
+	if err != nil {
+		if awsup.AWSErrorCode(err) == "InvalidNetworkInterfaceID.NotFound" {
+			// Concurrently deleted
+			return nil
+		}
+
+		if IsDependencyViolation(err) {
+			return err
+		}
+		return fmt.Errorf("error deleting ENI %q: %v", id, err)
```

Comment:
Since errors are usually printed prefixed with the word "error" I tend not to include the word "error" in the error string.